js/main.js

```diff
@@ -125,37 +127,94 @@ const addUpdateModal = function (isAdd, empId) { // Handle modal on update or ad
     const modalBackground = document.getElementById('modal-background');
     const modalContent = modalBackground.querySelector('#addUpdate-emp');
     const modalHeading = modalContent.querySelector('#modal-heading');
+    const skillDropdown = modalContent.querySelector('#addUpdate-skill-dropdown');
     const modalDropdown = modalContent.querySelector('.dropdown-element');
     const formSubmit = modalContent.querySelector('#form-submit');
     const formCancel = modalContent.querySelector('#form-cancel');
 
     modalHeading.innerText = (isAdd) ? 'Add Employee Details' : 'Update Employee Details';
     formSubmit.value = (isAdd) ? 'SAVE' : 'UPDATE';
 
-    if (!isAdd) {
-        fillUpdateClearForm(false, empId);
-    }
-
     modalBackground.classList.remove('display-none');
     modalContent.classList.remove('display-none');
 
+    skillDropdown.onclick = (event) => {
+        if (event.target.dataset.skillid.startsWith('id-')) {
+            const skillInputTag = document.getElementById('emp-skill');
+            const skillName = event.target.innerText;
+            let skillInput = document.getElementById('emp-skill').value.split(', ');
+
+            skillInput.push(skillName);
+
+            if (skillInput[0] == '') {
+                skillInput.shift()
+            }
+
+            skillInput = [...new Set(skillInput)]; // Remove duplicate elements in array
+
+            skillInputTag.value = skillInput.join(', ');
+            skillInputTag.focus();
+        }
+    }
+
     formCancel.onclick = () => {
         modalContent.classList.add('display-none');
         modalBackground.classList.add('display-none');
         if (!modalDropdown.querySelector('.display-none')) {
             toggleDropdown(modalDropdown);
         }
+        fillUpdateClearForm(true);
+    }
 
-        if (!isAdd) {
-            fillUpdateClearForm(true);
+    const displayErrorMessage = (isClear, validationErrorStatus) => {
+        const displayErrorSection = document.querySelectorAll('.validation-error');
+        displayErrorSection.forEach((val, index) => {
+            if (!isClear && validationErrorStatus[index] == 1) {
+                val.innerHTML = `<span>!Invalid Input!</span>`;
```

Comment:
@aromal-qburst Just "invalid input" will be enough

Reply:
Done